### ISSUE TITLE
Enable all backends to be tested in CI

### ISSUE THREAD

**Issue by hasezoey** [tramhao/termusic#212]
This issue requests / tracks the enablement of all backends to be tested / build in the CI. Currently only `rusty_backend` is build and tested.

Current roadblocks / considerations:

- not all packages are installed for the backends (though are mentioned via #211)
- mpv tests in CI will throw with a `VersionMismatch`, even though installed mpv is `0.34` and `0.29` being the minimal required (and working locally with `0.37`)
- mpv test `node_map` is flaky (at least locally); throwing Error `-10` (`The property exists, but is not available.`)

**Comment by tramhao** [comment]
The problem with testing is, right now CI use choco to install dependencies. In my own windows systems, I can only build in mingw64 of msys2. The building environment outside mingw64 is not easy to setup especially for gstreamer.

**Comment by hasezoey** [comment]
it is true that on windows it may be more complicated, but that should not hinder it being enabled for linux (or mac?) targets to test on once all the other roadblocks are dealt with

**Comment by tramhao** [comment]
Linux should have no problem. I stopped including libmpv myself, thus the test should be ok.

**Comment by tramhao** [comment]
Linux works. Macos doesn't work. Failed when brew install gstreamer. Windows doesn't change. 
Do you have experience installing gstreamer on macos?

**Comment by hasezoey** [comment]
>  Do you have experience installing gstreamer on macos?

i have absolutely no experience with macos, but from quickly looking at the logs, it seems like python fails to install / link because something in the chain already exists.
[reference](https://github.com/tramhao/termusic/actions/runs/8322489330/job/22770375022)

**Comment by tramhao** [comment]
Yes, I saw that as well. Basically several version of python3 were installed and have conflicts. As they were installed as dependency of gstreamer, there is not much thing I can do about it. For the time being, let's revert back ci for macos.
